### DISCUSSION TITLE
Seria a divisão entre frontend e backend um antipattern?

### DISCUSSION THREAD

**Discussion by woliveiras** [frontendbr/forum#1415] [Discussão]
Galera, li um artigo interessante e gostaria de ouvir (ler, no caso) a opinião de vocês sobre isso.

Eu tenho uma opinião formada, mas não quero enviesar sua resposta, então comenta ae o que pensa sobre esse assunto:

[Dividing frontend from backend is an antipattern](https://www.thoughtworks.com/insights/blog/dividing-frontend-backend-antipattern)

***P.S.**: muito provavelmente o autor não quis ofender ninguém com seu texto*

**Comment by Alecell** [comment]
Eu não li por cima, mas...

A divisão front e back nasceu decorrente de uma necessidade e não de uma mera vontade. 

A grande diferença do front e do back cuja não vi ser abordada na postagem é o HTML e o CSS. O front deve saber lidar com isso de forma semantica se preocupando com UI e UX. Este é um conhecimento diferenciado por si só.

O back e o front na parte "Quem programa de verdade" são iguais, complexidade de zonas são as mesmas hoje, mas em situações diferentes. (o back lida com um servidor legado e o front tem de dar suporte a IE8)

A grosso modo e ao meu ver, ambos são igualmente capazes como desenvolvedores, trabalham juntos para determinar como a aplicação vai prosseguir, entretanto, o back vai se preocupar com o servidor e o front com o browser.

**Comment by ghost** [comment]
não existe uma divisão real entre quem faz back-end e quem faz front-end, existem apenas linguagens diferentes. na prática ambos os profissionais trabalham juntos de forma quase simbiótica.

atualmente é comum desenvolvedores se identificarem como fullstack developer ou usarem frameworks isomórficos em javascript onde já não existe uma divisão clara entre o front-end e o back-end.

```
o back-end é ruim de dba e front-end
o dba é ruim de back-end e front-end
o front-end é ruim de back-end e dba
o fullstack é ruim de back-end, dba e front-end
Conclusão: todo mundo é ruim em alguma coisa
```

o problema está nos códigos legados e na especialização: a grande maioria dos sites ainda usa linguagens diferentes no front-end e no back-end, isso exige profissionais mais especializados em uma ou outra linguagem para manter uma boa qualidade de código e rapidez de desenvolvimento.

sou otimista, acredito que com o tempo as diferenças entre as linguagens tendem a diminuir assim como a distância entre o front e o back. tem muita coisa boa rolando no sentido de aproximar tanto as linguagens usadas no back do front quanto as linguagens usadas no front do back. 

esse negócio de fazer "flaming" e fingir que existe uma divisão clara quando na verdade não existe deve mesmo acabar com o tempo.

**Comment by ghost** [comment]
> entretanto, o back vai se preocupar com o servidor e o front com o browser.

@Alecell só por que é isso que é feito atualmente na maioria dos casos não significa que "vai" necessariamente continuar acontecendo. 

não existe front-end sem back-end, porém a reciproca não é verdadeira. um servidor sem back-end continua rodando sem problema, já um documento HTML sem servidor só abre localmente mesmo.

o mais provável é que com a popularização do SaaS o front-end ficará cada vez mais dependente do back-end e o back-end cada vez mais dependente do machine learning. 

no final ambos vão perder a guerra para o ML, mas o back-end deve ter uma morte bem mais lenta  do que o front-end que vai continuar sendo devorado vivo pelo back-end e pelo ML.
